hammers/scripts/clean_old_aggregates.py

```diff
@@ -63,21 +77,75 @@ def clear_aggregates(agg_list):
 
     return errors, report
 
+def orphan_find(allaggs):
+
+    # Find all hosts currently in aggregates
+    hosts_from_aggs = []
+    for agg in allaggs.values():
+        for host in agg['hosts']:
+            hosts_from_aggs.append(host)
+
+    # Make list of ironic hosts not in any aggregate
+    ironic_hosts = osrest.ironic.nodes(auth, details=False).keys()
+    orphans = []
+    for host in ironic_hosts:
+        if host not in hosts_from_aggs:
+            host_id = [h['id'] for h in osrest.blazar.hosts(auth).values() if h['uid'] == host][0]
+            orphans.append(host_id)
+    return(orphans)
+
+def has_active_allocation(orph):
+
+    matching_allocs = [alloc for alloc in host_allocs if alloc['resource_id'] == orph]
+    if not matching_allocs:
+        return False
+    res = matching_allocs[0]['reservations'][0]['id']
+    return(res)
+
+def del_expired_alloc(db, old_alloc):
+
+    ha_id = old_alloc['id']
+    l_id = old_alloc['lid']
+    hh = old_alloc['hypervisor_hostname']
+
+    msg=("Deleting host_allocation for host {} matching expired lease {}. ".format(hh,l_id) + "\n")
```

Comment:
Returning the error message feels a bit weird, as it's hard to parse for any other function that may want to call this later. Suggestion: return a tuple

```python
return hh, l_id
```

Then you can create the message like this (note I changed the tense to past tense)

```python
for alloc in old_allocations:
    hostname, lease_id = del_expired_alloc(conn, alloc)
    reports.append((
        "Deleted host_allocation for host {} matching expired lease {}.".format(hostname, lease_id))